jekyll/_cci2/orb-author-faq.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: classic-docs
+title: "Orb Author FAQ"
+short-title: "Orb Author FAQ"
+description: "Frequently asked questions from orb authors."
+categories: [configuring-jobs]
+order: 20
+---
+
+This document describes various questions and technical issues that you may find helpful when authoring orbs.
+
+* TOC
+{:toc}
+
+
+
+## Errors Claiming Namespace Or Publishing Orbs
+
+* Question: I recieve an error when attempting to claim a namespace or publish a production orb.
+
+* Answer: You may not be an organization owner/admin.
+
+Organizations may only claim a single namespace. In order to claim a namespace for an organization the authenticating user must possess owner/admin privledges over the organization. 
+
+If you do not have the proper permissions you may see an error similar to below:
+
+```
+Error: Unable to find organization YOUR_ORG_NAME of vcs-type GITHUB: Must have member permission.: the organization 'YOUR_ORG_NAME' under 'GITHUB' VCS-type does not exist. Did you misspell the organization or VCS?
+```
+
+Read more in our [Orbs Quickstart]({{site.baseurl}}/2.0/orb-author/#orbs-quickstart).
+
+
+## Secure API Tokens
+
+* Question: How do I protect user's API tokens from being commited to their config?
+
+* Answer: Utilize the "env_var_name" parameter type for the API key parameter. This parameter type will only accept POSIX valid environment variable name strings as valid. In the parameter description it is best to mention to the user to add this environment variable. 
+
+Read more:
+* [Environment Variable Name]({{site.baseurl}}/2.0/reusing-config/#environment-variable-name)
+* [Best Practices]({{site.baseurl}}/2.0/orbs-best-practices/)
+
+## Environment Variables
+
+* Question: How can I require a user to add an environment variable?
+* Answer: Create a parameter for the environment variable name, even if it is a statically named environment variable the user _should not_ change, and assign it the correct default value. In the parameter description let the user know if this value should not be changed. In either event instruct the user where they can obtain their API key. 
```

Comment:
I was thinking the required env vars could be documented and in the orb there could be a check done for the presence of the env vars. For example:

```
if [ -z "$MY_ENV_VAR" ]; then
  echo "Error: please set the value of the MY_ENV_VAR environment variable"
  exit 1
fi
```

Reply:
I think we can consider that a general good practice we should add the new orbs-best-practices and maybe link back to here. I'm going to open that PR, and then reference that here.

https://github.com/circleci/circleci-docs/pull/3773

^ Will update this PR with a link once this is merged.

Reply:
Added: https://github.com/circleci/circleci-docs/pull/3770/commits/3394e3cfda2369251fee584fdb256b1741bc8788